BaseClasses.py

```diff
@@ -766,9 +745,9 @@ def can_extend_magic(self, player: int, smallmagic: int = 16,
         elif self.has('Magic Upgrade (1/2)', player):
             basemagic = 16
         if self.can_buy_unlimited('Green Potion', player) or self.can_buy_unlimited('Blue Potion', player):
-            if self.world.item_functionality[player] == 'hard' and not fullrefill:
+            if self.world.item_functionality[player] == 2 and not fullrefill:
```

Comment:
looking later at this I will have no idea which one is 2 or 3, please compare with the option key variable or name